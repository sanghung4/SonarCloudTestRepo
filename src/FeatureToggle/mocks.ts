import { Feature } from 'generated/graphql';

export const mockFeatures: Feature[] = [
  { id: '1', isEnabled: false, name: 'noooo' },
  { id: '2', isEnabled: true, name: 'fine' },
  { id: '3', isEnabled: true, name: 'test' }
];
