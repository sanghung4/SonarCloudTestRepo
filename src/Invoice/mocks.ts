import { GetInvoiceDocument } from 'generated/graphql';

export const successMincron = {
  request: {
    query: GetInvoiceDocument,
    variables: {
      accountId: '123456',
      invoiceNumber: '5460026'
    }
  },
  result: {
    data: {
      invoice: {
        invoiceNumber: '5460026',
        status: 'Open',
        terms: 'NET 30 DAYS',
        customerPo: '40251/13325',
        invoiceDate: '10/28/2020',
        dueDate: '10/28/2020',
        originalAmt: '',
        openBalance: '',
        age: '',
        jobNumber: 'AVLNEP7',
        jobName: 'NEP 7 ASHEVILLE',
        address: {
          streetLineOne: '911 BLUEBERRY WAY',
          streetLineTwo: '',
          streetLineThree: '',
          city: 'NORTHLAKE',
          state: 'TX',
          postalCode: '76226',
          country: ''
        },
        shipDate: '11/06/2020',
        deliveryMethod: 'Our Truck',
        subtotal: '3184.28',
        tax: '262.76',
        otherCharges: '0.0',
        paidToDate: '1000.00',
        invoiceItems: [
          {
            id: '1738433F4-91A7-4637-8C60-B4DF59A060C1',
            brand: 'Advanced Drainage Systems, Inc.',
            name: '120 x 7.5, 100 Sq Yd, Geosynthetic, Double, Photodegradable, Erosion Control Blanket',
            partNumber: '8X1125STRAWDN',
            mfr: '00S2TT',
            thumb: 'null',
            price: '43.59',
            qty: {
              quantityOrdered: 1,
              quantityShipped: 1
            }
          }
        ]
      }
    }
  }
};

export const successMincron_5448540 = {
  request: {
    query: GetInvoiceDocument,
    variables: {
      accountId: '91492',
      invoiceNumber: '5448540'
    }
  },
  result: {
    data: {
      invoice: {
        invoiceNumber: '5282411',
        status: 'Open',
        terms: 'NET 30 DAYS',
        customerPo: 'WILKESBORO JIM',
        invoiceDate: '10/06/2021',
        dueDate: '10/28/2020',
        originalAmt: '',
        openBalance: '',
        age: '',
        jobNumber: 'SHOP',
        jobName: 'SHOP',
        address: {
          streetLineOne: '911 BLUEBERRY WAY',
          streetLineTwo: '',
          streetLineThree: '',
          city: 'NORTHLAKE',
          state: 'TX',
          postalCode: '76226',
          country: ''
        },
        shipDate: '11/06/2021',
        deliveryMethod: 'Our Truck',
        subtotal: '2184.28',
        tax: '162.73',
        otherCharges: '0.0',
        paidToDate: '139.91',
        invoiceItems: {
          id: 'F02FD85F-4DCE-4EA7-8459-2DA7CFD689DC',
          brand: 'Certex USA, name:2 x 12, Nylon, Eye and Eye, Ply Sling',
          partNumber: 'SLING212,mfr:CX07-0126-12',
          thumb:
            'https://images.tradeservice.com/ProductImages/DIR100193/CERTEX_CX07-0126-12_LINE_SML.jpg',
          price: '40.00',
          qty: {
            quantityOrdered: 1,
            quantityShipped: 1
          }
        }
      }
    }
  }
};
