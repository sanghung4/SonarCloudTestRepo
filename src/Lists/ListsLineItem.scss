@use 'src/_reece.scss';

// default (usually mobile)
.lists-line-item {
  &__container {
    position: relative;
    margin: 0;
    padding-top: 8px;
    padding-bottom: 8px;
    width: 100%;
    height: 99%;
    flex-wrap: wrap;
    align-items: center;
  }
  &__loader {
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
  }
  &__product {
    &--details-block {
      padding: 0 16px;
      width: 100%;
    }
    &--image-subcontainer {
      padding: 0;
      max-width: 80px;
      min-height: 80px;
      display: flex;
      align-items: center;
      justify-content: center;
      & a {
        display: flex;
        align-items: center;
        justify-content: center;
      }
    }
    &--details-container {
      display: flex;
      flex-wrap: wrap;
      align-items: center;
      align-content: flex-start;
    }
    &--mfr-name {
      color: reece.$dark-text-gray-80;
      font-size: 0.7rem;
      line-height: 0.9rem;
      text-transform: uppercase;
      & .MuiSkeleton-root {
        width: 100%;
      }
    }
    &--item-name {
      margin: 8px 0;
      color: reece.$dark-blue-100;
      font-size: 1rem;
      line-height: 1.2rem;
      letter-spacing: 0.005em;
      & .MuiSkeleton-root {
        width: 100%;
        height: 100%;
        margin-bottom: 4px;
      }
    }
    &--mfr-msc {
      color: reece.$dark-text-gray-80;
      font-size: 0.7rem;
      line-height: 0.9rem;
      text-transform: uppercase;
      padding-right: 8px;
      display: block;
      & .MuiSkeleton-root {
        width: 100%;
      }
    }
    &--branch-available {
      &--container {
        padding: 0;
        display: flex;
        justify-content: flex-start;
        align-items: center;
        flex-wrap: nowrap;
      }
      &--available-chip-box {
        overflow: hidden;
        text-align: center;
        font-size: 0.7rem;
      }
      &--location-box {
        margin-left: 8px;
        .MuiButton-startIcon {
          width: 12px;
        }
        & button {
          font-weight: 700;
          font-size: 0.7rem;
        }
      }
    }
    &--mobile-price-per-unit {
      font-weight: 500;
      font-size: 1rem;
      line-height: 1rem;
      letter-spacing: 0.005em;
      color: reece.$dark-blue-100;
      padding: 8px 16px;
    }
  }
  &__reoder-handle-container {
    display: none;
  }
  &__image-container {
    display: flex;
    align-items: flex-start;
    justify-content: center;
    flex-direction: column;
  }
  &__hidden-print-block {
    display: none;
  }
  &__mobile-delete-box {
    position: absolute;
    top: 0;
    right: 0;
  }
}
// Desktop
@media (min-width: reece.$breakpoint-md) {
  .lists-line-item {
    &__container {
      flex-wrap: nowrap;
    }
    &__product {
      &--image-subcontainer {
        width: 100px;
        height: 100px;
        & a {
          width: 100%;
          height: 100%;
        }
      }
      &--details-container {
        align-content: initial;
      }
      &--mfr-name {
        font-size: 0.8rem;
        line-height: 1.0125rem;
        & .MuiSkeleton-root {
          width: 100px;
        }
      }
      &--item-name {
        font-size: 1rem;
        line-height: 1.5rem;
        letter-spacing: 0.005em;
        margin: 0 0;
        & .MuiSkeleton-root {
          width: 400px;
        }
      }
      &--mfr-msc {
        font-size: 0.8rem;
        line-height: 1.0125rem;
        padding-right: 32px;
        display: initial;
        & .MuiSkeleton-root {
          width: 100px;
        }
      }
      &--branch-available {
        &--container {
          padding: 8px 16px;
          width: 100%;
          font-size: 0.8rem;
        }
        &--available-chip-box {
          overflow: hidden;
          text-align: center;
          font-size: 0.8rem;
        }
        &--location-box {
          margin-left: 18px;
          .MuiButton-startIcon {
            width: 16px;
          }
          & button {
            font-size: 0.8rem;
          }
        }
      }
    }
    &__image-container {
      align-items: center;
    }
    &__reoder-handle-container {
      display: flex;
      align-items: center;
      justify-content: center;
      padding: 0 16px 0 8px;
    }
  }
}

// Print
@media print {
  .lists-line-item {
    &__hidden-print-block {
      display: block;
    }
    &__product {
      &--branch-available {
        &--container {
          display: none;
        }
      }
    }
  }
}
