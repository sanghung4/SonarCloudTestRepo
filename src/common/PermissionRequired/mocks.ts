import { UserContextType } from 'AuthProvider';
import { Permission } from 'common/PermissionRequired';

export const authConfigNoPerm = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
export const authConfig = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    userId: 'test',
    permissions: [Permission.EDIT_LIST, Permission.APPROVE_CART],
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
