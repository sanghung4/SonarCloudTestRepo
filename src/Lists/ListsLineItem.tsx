import { useCallback, useContext, useMemo } from 'react';

import {
  Grid,
  Hidden,
  IconButton,
  Image,
  Link,
  Skeleton,
  useScreenSize
} from '@dialexa/reece-component-library';
import { DraggableProvidedDragHandleProps } from 'react-beautiful-dnd';
import Dotdotdot from 'react-dotdotdot';
import { useTranslation } from 'react-i18next';
import { Link as RouterLink } from 'react-router-dom';
import slugify from 'react-slugify';

import FeedbackButton from 'common/FeedbackButton';
import Loader from 'common/Loader';
import QtyInput from 'common/QtyInput';
import { ListLineItem, ProductPricing } from 'generated/graphql';
import { DeleteBinIcon, HandleIcon, WarningIcon } from 'icons';
import notfound from 'images/notfound.png';
import 'Lists/ListsLineItem.scss';
import { BranchContext } from 'providers/BranchProvider';
import { useCartContext } from 'providers/CartProvider';
import AvailabilityChip from 'Product/AvailabilityChip';
import AdvancedToolTip from 'common/AdvancedToolTip';
import { useListsContext } from 'providers/ListsProvider';
import { format } from 'utils/currency';
import ListLineItemAvailability from 'Lists/ListLineItemAvailability';
import CustomerPartNumber from 'Search/CustomerPartNumber';

/**
 * Config
 */
const NOT_AVAILABLE_STATS = [
  'Delete',
  'Purge',
  'NonPDW',
  'Discontinued',
  'Comment',
  'NotInElastic',
  'NotInEclipse'
];

/**
 * Types
 */
type Props = {
  item: ListLineItem;
  loading?: boolean;
  disabled?: boolean;
  updateItem?: (item: ListLineItem) => void;
  priceData?: ProductPricing;
  priceDataLoading?: boolean;
  dragHandleProps?: DraggableProvidedDragHandleProps;
};

/**
 * Component
 */
function ListsLineItem(props: Props) {
  /**
   * Props
   */
  const {
    item,
    loading,
    disabled,
    updateItem,
    priceData,
    priceDataLoading,
    dragHandleProps
  } = props;

  /**
   * Custom hooks
   */
  const { t } = useTranslation();
  const { isSmallScreen } = useScreenSize();

  /**
   * Memos
   */
  const notAvailable = useMemo(notAvailableMemo, [
    priceData,
    priceDataLoading,
    item.status
  ]);

  /**
   * Context
   */
  const { addItemToCart, disableAddToCart } = useCartContext();
  const {
    availableListIds,
    isUpdating,
    toggleItemFromList,
    selectedList,
    setAvailableListIds,
    setSelectedList
  } = useListsContext();
  const { shippingBranch } = useContext(BranchContext);

  /**
   * Callbacks
   */
  const deleteItem = useCallback(deleteItemCb, [
    availableListIds,
    item?.erpPartNumber,
    item.id,
    selectedList,
    setAvailableListIds,
    setSelectedList,
    toggleItemFromList
  ]);
  const handleAddToCart = () => {
    addItemToCart(item.erpPartNumber ?? '', item?.quantity ?? 0, 0, priceData);
  };
  const updateQty = (newValue: number) => {
    updateItem?.({ ...props.item, quantity: newValue } as ListLineItem);
  };

  /**
   * Misc
   */
  // URL
  const toSlug = `${item.manufacturerName}-${item.name}-${item.manufacturerNumber}`;
  const productSlug = slugify(toSlug);
  const productUrl = `/product/${productSlug}/MSC-${item?.erpPartNumber}`;
  // Display
  const uom = priceData?.orderUom?.toLowerCase() ?? t('product.each');
  const pricePerUnit = `${format(priceData?.sellPrice ?? 0)} ${uom}`;
  const manufacturerNumber = item?.manufacturerNumber
    ? `${t('product.mfr')} ${item?.manufacturerNumber}`
    : '-';
  const msc = item?.erpPartNumber
    ? `${t('product.msc')}${item.erpPartNumber}`
    : '-';
  // CSS Util
  const cn = (name: string) => 'lists-line-item' + name;

  /**
   * Render
   */
  return (
    <Grid container className={cn('__container')}>
      {/* üîµ Loader */}
      {isUpdating === item?.erpPartNumber && priceData === undefined && (
        <Grid item className={cn('__loader')}>
          <Loader backdrop />
        </Grid>
      )}
      {/* üîµ CHUNK 1 - Reorder Handle, Product Image, Info */}
      <Grid
        container
        item
        xs={12}
        md={7} // UPDATE THIS
        py={0}
        {...dragHandleProps}
        flexWrap="nowrap"
      >
        {/* Reorder Handle */}
        <Grid item xs="auto" className={cn('__reoder-handle-container')}>
          {!disabled && <HandleIcon />}
        </Grid>
        {/* Product Image */}
        <Grid item xs="auto" className={cn('__image-container')}>
          <div
            className={cn('__product--image-subcontainer')}
            data-testid={`product-picture-${item?.erpPartNumber}`}
          >
            {loading ? (
              <Skeleton variant="rectangular" height="100%" width="100%" />
            ) : (
              <Link to={productUrl} component={RouterLink}>
                <Image
                  alt={t('common.productPicture')}
                  fallback={notfound}
                  src={item?.imageUrls?.thumb ?? ''}
                />
              </Link>
            )}
          </div>
        </Grid>
        {/* Product Info */}
        <Grid item xs className={cn('__product--details-container')}>
          {/* Availability */}
          <div className={cn('__hidden-print-block')}>
            <AvailabilityChip
              branch={shippingBranch}
              loading={Boolean(priceDataLoading)}
              stock={
                shippingBranch?.isPricingOnly
                  ? priceData?.totalAvailableQty
                  : priceData?.branchAvailableQty
              }
            />
          </div>
          {/* Main Product Details Container*/}
          <div className={cn('__product--details-block')}>
            {/* Manufacturer name */}
            <span
              className={cn('__product--mfr-name')}
              data-testid={`manufacturer-name-${item?.erpPartNumber}`}
            >
              {loading ? (
                <Skeleton data-testid="loading-manufacturer-name" />
              ) : (
                item?.manufacturerName ?? '-'
              )}
            </span>
            {/* Item name */}
            <div className={cn('__product--item-name')}>
              <span data-testid={`item-name-${item?.erpPartNumber}`}>
                {loading ? (
                  <>
                    <Skeleton />
                    <Hidden mdUp>
                      <Skeleton />
                    </Hidden>
                  </>
                ) : (
                  <Dotdotdot clamp={2}>
                    <Link to={productUrl} component={RouterLink}>
                      {item?.name ?? '-'}
                    </Link>
                  </Dotdotdot>
                )}
              </span>
            </div>
            {/* MFR */}
            <span
              className={cn('__product--mfr-msc')}
              data-testid={`manufacturer-number-${item?.erpPartNumber}`}
            >
              {loading ? <Skeleton /> : manufacturerNumber}
            </span>
            {/* MSC */}
            <span
              className={cn('__product--mfr-msc')}
              data-testid={`product-id-${item?.erpPartNumber}`}
            >
              {loading ? <Skeleton /> : msc}
            </span>
            {loading ? (
              <Skeleton />
            ) : (
              props.item.customerPartNumber && (
                <CustomerPartNumber
                  partNumber={props.item.customerPartNumber}
                  stylingForPage="MyLists"
                />
              )
            )}
          </div>
          {/* [Desktop üñ•Ô∏è] Availability and branch */}
          <Hidden mdDown>
            <ListLineItemAvailability notAvailable={notAvailable} {...props} />
          </Hidden>
          {/* [MOBILE üì±] Price per unit */}
          <Hidden mdUp>
            <div>
              <span
                className={cn('__product--mobile-price-per-unit')}
                data-testid={`price-per-unit-${item?.erpPartNumber}`}
              >
                {loading || priceDataLoading ? (
                  <Skeleton
                    width={60}
                    data-testid={`price-skeleton-${item?.erpPartNumber}`}
                  />
                ) : (
                  pricePerUnit
                )}
              </span>
            </div>
          </Hidden>
        </Grid>
        {/* [MOBILE üì±] Delete button */}
        <Hidden mdUp>
          <div className={cn('__mobile-delete-box')}>
            <IconButton
              color="primary"
              disabled={Boolean(
                loading ||
                  isUpdating === item?.erpPartNumber ||
                  priceDataLoading
              )}
              onClick={deleteItem}
              data-testid={`delete-button-${item?.erpPartNumber}`}
              size="large"
              sx={{ displayPrint: 'none' }}
            >
              <DeleteBinIcon />
            </IconButton>
          </div>
        </Hidden>
      </Grid>
      {/* üîµ MOBILE BONUS CHUNK üì± -  Availability and branch */}
      <Hidden mdUp>
        <Grid xs={12}>
          <ListLineItemAvailability notAvailable={notAvailable} {...props} />
        </Grid>
      </Hidden>
      {/* üîµ CHUNK 2 - Pricing, Qty, Add to Cart, Delete */}
      <Grid
        container
        item
        xs={12}
        md={5}
        spacing={1}
        alignItems="center"
        justifyContent="flex-start"
        wrap="nowrap"
        direction="row"
      >
        {/* [Desktop üñ•Ô∏è] Price per unit */}
        <Hidden mdDown>
          <Grid item xs={4} textAlign="center">
            <span data-testid={`price-per-unit-${item?.erpPartNumber}`}>
              {loading || priceDataLoading ? (
                <Skeleton
                  width={60}
                  data-testid={`price-skeleton-${item?.erpPartNumber}`}
                />
              ) : (
                pricePerUnit
              )}
            </span>
          </Grid>
        </Hidden>
        {/* Qty input */}
        <Grid item xs={6} md="auto">
          {loading ? (
            <Skeleton width={125} height={37} variant="rectangular" />
          ) : (
            <QtyInput
              value={item?.quantity || item?.minIncrementQty || 1}
              onUpdate={updateQty}
              max={9999}
              increment={item?.minIncrementQty || 1}
              disabled={
                isUpdating === item?.erpPartNumber || loading || notAvailable
              }
              data-testid="list-line-item-qty-input"
              size="small"
              noDebounce
            />
          )}
        </Grid>
        {/* Add to Cart */}
        <Grid item xs={6} md="auto">
          <AdvancedToolTip
            title="Warning"
            text={t('cart.maxLimitToolTip')}
            icon={<WarningIcon width={16} height={16} />}
            placement="bottom"
            disabled={disableAddToCart && !isSmallScreen}
          >
            <FeedbackButton
              fullWidth
              testId={`${item?.id}-add-to-cart-button`}
              onClick={handleAddToCart}
              disabled={
                loading || priceDataLoading || notAvailable || disableAddToCart
              }
              value={t('common.addToCart')}
              valueDone={t('common.addedToCart')}
              sx={{ displayPrint: 'none', height: '37px' }}
            />
          </AdvancedToolTip>
        </Grid>
        {/* [Desktop üñ•Ô∏è] Delete button */}
        <Hidden mdDown>
          <Grid item xs="auto" sx={{ displayPrint: 'none' }}>
            <IconButton
              color="primary"
              onClick={deleteItem}
              data-testid={`delete-button-${item?.erpPartNumber}`}
              disabled={
                loading ||
                isUpdating === item?.erpPartNumber ||
                priceDataLoading
              }
              size="small"
            >
              <DeleteBinIcon />
            </IconButton>
          </Grid>
        </Hidden>
      </Grid>
      {/* Pricing - END */}
    </Grid>
  );

  /**
   * Callback Defs
   */
  function deleteItemCb() {
    if (!selectedList) {
      return;
    }

    // Filter List
    const updatedListLineItems = selectedList.listLineItems.filter(
      ({ id }) => id !== item.id
    );
    setSelectedList({ ...selectedList, listLineItems: updatedListLineItems });

    // Remove the current listId from item's availableListIds
    const listIds = availableListIds.filter(
      (listId) => listId !== selectedList.id
    );
    setAvailableListIds(listIds);

    // Call
    toggleItemFromList?.(item?.erpPartNumber ?? '');
  }

  /**
   * Memo Defs
   */
  function notAvailableMemo() {
    const status =
      !priceData && !priceDataLoading ? 'NotInEclipse' : item.status;

    return NOT_AVAILABLE_STATS.includes(status ?? '');
  }
}

export default ListsLineItem;
