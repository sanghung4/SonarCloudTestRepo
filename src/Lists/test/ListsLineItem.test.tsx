import ListsLineItem from 'Lists/ListsLineItem';
import mocks, {
  COPPER_CLAMPS_ADD_TO_CART,
  LIST_PROVIDER_MOCKS
} from 'Lists/test/index.mocks';
import * as t from 'locales/en/translation.json';
import { setBreakpoint } from 'test-utils/mockMediaQuery';
import { render } from 'test-utils/TestWrapper';

describe('ListsLineItem', () => {
  it('Should match snapshot when loading', () => {
    const { container } = render(
      <ListsLineItem
        loading
        item={LIST_PROVIDER_MOCKS.VALID_LIST.selectedList?.listLineItems[0]!}
      />
    );
    expect(container).toMatchSnapshot();
  });

  // TO-DO - FIX THIS LATER
  it.skip('Should display find at other branches', async () => {
    const { findAllByTestId, findByTestId } = render(
      <ListsLineItem
        item={LIST_PROVIDER_MOCKS.VALID_LIST.selectedList?.listLineItems[0]!}
      />
    );
    const branchButton = await findAllByTestId('find-at-other-branches-button');

    expect(branchButton[0]).toBeInTheDocument();
    expect(branchButton[0]).toHaveTextContent(t.product.checkNearByBranches);
    expect(branchButton[0]).toMatchSnapshot();
  });

  it('Should display the availability of an item in the list', async () => {
    const { findAllByTestId } = render(
      <ListsLineItem
        item={LIST_PROVIDER_MOCKS.VALID_LIST.selectedList?.listLineItems[0]!}
      />
    );

    const stockChip = await findAllByTestId('availability-stock-chip');

    expect(stockChip[0]).toHaveTextContent(t.common.availableForOrder);
    expect(stockChip[0]).toMatchSnapshot();
  });

  it('Should match snapshot for mobile view', () => {
    setBreakpoint('mobile');
    const { container } = render(
      <ListsLineItem
        loading
        priceDataLoading
        item={LIST_PROVIDER_MOCKS.VALID_LIST.selectedList?.listLineItems[0]!}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
