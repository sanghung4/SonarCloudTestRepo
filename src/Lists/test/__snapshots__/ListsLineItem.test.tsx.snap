// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListsLineItem Should display find at other branches 1`] = `
<button
  class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-disableElevation MuiButtonBase-root  css-1p51lwh-MuiButtonBase-root-MuiButton-root"
  data-testid="find-at-other-branches-button"
  tabindex="0"
  type="button"
>
  Find at Other Branches
</button>
`;

exports[`ListsLineItem Should display the availability of an item in the list 1`] = `
<div
  class="MuiBox-root css-kpy16r"
  data-testid="availability-stock-chip"
  title="Available for order"
>
  <svg>
    quick-order.svg
  </svg>
   Available for order
</div>
`;

exports[`ListsLineItem Should match snapshot for mobile view 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container lists-line-item__container css-11lq3yg-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-7 css-9runz6-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto lists-line-item__reoder-handle-container css-1tsghma-MuiGrid-root"
      >
        <svg>
          handle.svg
        </svg>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto lists-line-item__image-container css-1tsghma-MuiGrid-root"
      >
        <div
          class="lists-line-item__product--image-subcontainer"
          data-testid="product-picture-33860"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3x7rqn-MuiSkeleton-root"
            style="width: 100%; height: 100%;"
          />
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true lists-line-item__product--details-container css-1z0dvx8-MuiGrid-root"
      >
        <div
          class="lists-line-item__hidden-print-block"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-q9xsaf-MuiSkeleton-root"
            style="width: 100%; height: 24px;"
          />
        </div>
        <div
          class="lists-line-item__product--details-block"
        >
          <span
            class="lists-line-item__product--mfr-name"
            data-testid="manufacturer-name-33860"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
              data-testid="loading-manufacturer-name"
            />
          </span>
          <div
            class="lists-line-item__product--item-name"
          >
            <span
              data-testid="item-name-33860"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
              />
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
              />
            </span>
          </div>
          <span
            class="lists-line-item__product--mfr-msc"
            data-testid="manufacturer-number-33860"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
            />
          </span>
          <span
            class="lists-line-item__product--mfr-msc"
            data-testid="product-id-33860"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
            />
          </span>
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
          />
        </div>
        <div>
          <span
            class="lists-line-item__product--mobile-price-per-unit"
            data-testid="price-per-unit-33860"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
              data-testid="price-skeleton-33860"
              style="width: 60px;"
            />
          </span>
        </div>
      </div>
      <div
        class="lists-line-item__mobile-delete-box"
      >
        <button
          class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorPrimary MuiIconButton-sizeLarge css-19dmyy1-MuiButtonBase-root-MuiIconButton-root"
          data-testid="delete-button-33860"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <svg>
            delete-bin.svg
          </svg>
        </button>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-grid-xs-12 css-1634xdp-MuiGrid-root"
    >
      <div
        class="lists-line-item__product--branch-available--container"
      >
        <div
          class="lists-line-item__product--branch-available--available-chip-box"
          data-testid="stock-chip-box"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-q9xsaf-MuiSkeleton-root"
            style="width: 100%; height: 24px;"
          />
        </div>
        <div
          class="lists-line-item__product--branch-available--location-box"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
          />
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-1 MuiGrid-wrap-xs-nowrap MuiGrid-grid-xs-12 MuiGrid-grid-md-5 css-5vb1iz-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-auto css-1xh0yc4-MuiGrid-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3x7rqn-MuiSkeleton-root"
          style="width: 125px; height: 37px;"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-auto css-1xh0yc4-MuiGrid-root"
      >
        <div
          aria-label=""
          class=""
          data-mui-internal-clone-element="true"
          data-testid="tipMobile"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-fullWidth MuiButtonBase-root Mui-disabled  css-dpvwd0-MuiButtonBase-root-MuiButton-root"
            data-testid="1bd15d80-fb49-4410-965a-0f212374ac3f-add-to-cart-button"
            disabled=""
            tabindex="-1"
            type="button"
          >
            Add to Cart
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ListsLineItem Should match snapshot when loading 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container lists-line-item__container css-11lq3yg-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-7 css-9runz6-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto lists-line-item__reoder-handle-container css-1tsghma-MuiGrid-root"
      >
        <svg>
          handle.svg
        </svg>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto lists-line-item__image-container css-1tsghma-MuiGrid-root"
      >
        <div
          class="lists-line-item__product--image-subcontainer"
          data-testid="product-picture-33860"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3x7rqn-MuiSkeleton-root"
            style="width: 100%; height: 100%;"
          />
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true lists-line-item__product--details-container css-1z0dvx8-MuiGrid-root"
      >
        <div
          class="lists-line-item__hidden-print-block"
        >
          <div
            class="MuiBox-root css-kpy16r"
            data-testid="availability-stock-chip"
            title="Available for order"
          >
            <svg>
              quick-order.svg
            </svg>
             Available for order
          </div>
        </div>
        <div
          class="lists-line-item__product--details-block"
        >
          <span
            class="lists-line-item__product--mfr-name"
            data-testid="manufacturer-name-33860"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
              data-testid="loading-manufacturer-name"
            />
          </span>
          <div
            class="lists-line-item__product--item-name"
          >
            <span
              data-testid="item-name-33860"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
              />
            </span>
          </div>
          <span
            class="lists-line-item__product--mfr-msc"
            data-testid="manufacturer-number-33860"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
            />
          </span>
          <span
            class="lists-line-item__product--mfr-msc"
            data-testid="product-id-33860"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
            />
          </span>
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
          />
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-1 MuiGrid-wrap-xs-nowrap MuiGrid-grid-xs-12 MuiGrid-grid-md-5 css-5vb1iz-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-auto css-1xh0yc4-MuiGrid-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3x7rqn-MuiSkeleton-root"
          style="width: 125px; height: 37px;"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-auto css-1xh0yc4-MuiGrid-root"
      >
        <div
          aria-label=""
          class=""
          data-mui-internal-clone-element="true"
          data-testid="tipDesktop"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-fullWidth MuiButtonBase-root Mui-disabled  css-dpvwd0-MuiButtonBase-root-MuiButton-root"
            data-testid="1bd15d80-fb49-4410-965a-0f212374ac3f-add-to-cart-button"
            disabled=""
            tabindex="-1"
            type="button"
          >
            Add to Cart
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
